File: JAVA_APPLICATION/app/src/main/java/com/example/myapplication/PromotionsAdapter.java
package com.example.myapplication;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class PromotionsAdapter extends RecyclerView.Adapter<PromotionsAdapter.PromotionViewHolder> {

    private List<Promotion> promotions;

    public PromotionsAdapter(List<Promotion> promotions) {
        this.promotions = promotions;
    }

    public static class PromotionViewHolder extends RecyclerView.ViewHolder {
        public TextView promoTitle;
        public TextView promoDescription;
        public TextView promoDiscount;

        public PromotionViewHolder(View itemView) {
            super(itemView);
            promoTitle = itemView.findViewById(R.id.textPromoTitle);
            promoDescription = itemView.findViewById(R.id.textPromoDescription);
            promoDiscount = itemView.findViewById(R.id.textPromoDiscount);
        }
    }

    @Override
    public PromotionViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_promotion, parent, false);
        return new PromotionViewHolder(view);
    }

    @Override
    public void onBindViewHolder(PromotionViewHolder holder, int position) {
        Promotion promotion = promotions.get(position);
        holder.promoTitle.setText(promotion.getTitle());
        holder.promoDescription.setText(promotion.getDescription());
        holder.promoDiscount.setText("Discount: " + promotion.getDiscount());
    }

    @Override
    public int getItemCount() {
        return promotions.size();
    }
}

File: app/src/main/java/com/example/city_cycle_app/PromotionsAdapter.kt
package com.example.city_cycle_app

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class PromotionsAdapter(private val promotions: List<Promotion>) :
    RecyclerView.Adapter<PromotionsAdapter.PromotionViewHolder>() {

    class PromotionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val promoTitle: TextView = itemView.findViewById(R.id.textPromoTitle)
        val promoDescription: TextView = itemView.findViewById(R.id.textPromoDescription)
        val promoDiscount: TextView = itemView.findViewById(R.id.textPromoDiscount)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PromotionViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_promotion, parent, false)
        return PromotionViewHolder(view)
    }

    override fun onBindViewHolder(holder: PromotionViewHolder, position: Int) {
        val promotion = promotions[position]
        holder.promoTitle.text = promotion.title
        holder.promoDescription.text = promotion.description
        holder.promoDiscount.text = "Discount: ${promotion.discount}"
    }

    override fun getItemCount() = promotions.size
}
